src/conn/conn_api.c

```diff
@@ -1853,16 +1876,21 @@ __conn_single(WT_SESSION_IMPL *session, const char *cfg[])
     }
 
     /*
-     * We own the database home, figure out if we're creating it. There are a few files created when
-     * initializing the database home and we could crash in-between any of them, so there's no
-     * simple test. The last thing we do during initialization is rename a turtle file into place,
-     * and there's never a database home after that point without a turtle file. If the turtle file
-     * doesn't exist, it's a create.
+     * If WiredTiger file exists but is size zero when it is not supposed to be (the turtle file
+     * exists and we are not salvaging), write a message but don't fail.
      */
-    WT_ERR(__wt_turtle_exists(session, &exist));
-    conn->is_new = exist ? 0 : 1;
+    empty = false;
+    if (fh) {
```

Comment:
It seems that it can be NULL, see my [comment](https://github.com/wiredtiger/wiredtiger/pull/7388#issuecomment-1005664261). We would need to dig deeper into this test to understand why.

Reply:
I understand now, it seems that the check, allows fh to be null. 
``` if (F_ISSET(conn, WT_CONN_READONLY)) {
        if (ret == EACCES || ret == ENOENT)
            ret = 0;
        WT_ERR(ret);```
in that case with can you change `(fh)` to `(fh != NULL)` or place it the above else statement?